### PR TITLE
added some test form repository service, controller

### PR DESCRIPTION
**Pull request by DevIgork**

### CODE REVIEW

File: src/test/resources/database/shoppingCart/delete_all_cart_data.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+DELETE FROM shopping_carts;
```
Comment (DEz-1996) [inline src/test/resources/database/shoppingCart/delete_all_cart_data.sql:1]: Add empty line)

File: src/test/resources/database/shoppingCart/insert_into_shopping_carts.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO shopping_carts (user_id)
+VALUES (1);
```
Comment (DEz-1996) [inline src/test/resources/database/shoppingCart/insert_into_shopping_carts.sql:2]: Add empty line)

**Review by DEz-1996 — APPROVED** [review]
LGFM

File: src/test/java/com/example/bookstore/controller/ShoppingCartControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        cleanCoder.setDescription(CLEAN_CODER_DESCRIPTION);
+        cleanCoder.setIsbn(CLEAN_CODER_ISBN);
+        cleanCoder.setAuthor(CLEAN_CODER_AUTHOR);
+        cleanCoder.setCoverImage(CLEAN_CODER_COVER_IMAGE);
+    }
+
+    @WithMockUser(username = "user", roles = {"USER"})
+    @Test
+    @DisplayName("""
+            add to cart
```
Comment (yevheniisynytsia) [inline src/test/java/com/example/bookstore/controller/ShoppingCartControllerTest.java]:
```suggestion
            Add to cart, valid data
```
let's make names in such style to be more informative

File: src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    private static final String CLEAN_CODER_AUTHOR = "Robert C. Martin";
+    private static final String CLEAN_CODER_ISBN =
+            "9789669820229";
+    private static final String CLEAN_CODER_DESCRIPTION =
+            "A Code of Conduct for Professional Programmers,"
+                    + " legendary software expert Robert C. Martin";
+    private static final String CLEAN_CODER_COVER_IMAGE = "https://balka-book.com/files/2023/07_06/12_56/u_files_store_5_7.jpg";
+    private static Book cleanCoder;
+    @InjectMocks
+    private ShoppingCartServiceImpl cartService;
```
Comment (yevheniisynytsia) [inline src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java]: usually mock are defined first and then the service that you are testing with `@InjectMocks`

File: src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        cleanCoder.setId(CLEAN_CODER_ID);
+        cleanCoder.setDescription(CLEAN_CODER_DESCRIPTION);
+        cleanCoder.setIsbn(CLEAN_CODER_ISBN);
+        cleanCoder.setAuthor(CLEAN_CODER_AUTHOR);
+        cleanCoder.setCoverImage(CLEAN_CODER_COVER_IMAGE);
+    }
+
+    @Test
+    @DisplayName("""
+            verify addToCart() method work
```
Comment (yevheniisynytsia) [inline src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java]: Add  cart item to shopping cart, valid data

File: src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        cartItem.setBook(cleanCoder);
+        cartItem.setShoppingCart(shoppingCart);
+        CartItemDto cartItemDto = new CartItemDto()
+                .setBookId(cleanCoder.getId())
+                .setBookTitle(cleanCoder.getTitle())
+                .setQuantity(1);
+        ShoppingCartDto shoppingCartDto = new ShoppingCartDto()
+                .setCartItem(List.of(cartItemDto))
+                .setUserId(1L);
+        when(bookRepository.findById(1L)).thenReturn(Optional.ofNullable(cleanCoder));
```
Comment (yevheniisynytsia) [inline src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java:95]:
```suggestion

        when(bookRepository.findById(1L)).thenReturn(Optional.ofNullable(cleanCoder));
```

File: src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        when(userRepository.findById(1L)).thenReturn(Optional.of(user));
+        when(cartRepository.findByUserId(1L)).thenReturn(Optional.of(shoppingCart));
+        when(cartItemRepository.save(cartItem)).thenReturn(cartItem);
+        when(cartMapper.toDto(shoppingCart)).thenReturn(shoppingCartDto);
+        ShoppingCartDto actual = cartService.addToCart(requestDto, 1L);
```
Comment (yevheniisynytsia) [inline src/test/java/com/example/bookstore/service/ShoppingCartServiceTest.java:101]:
```suggestion

        ShoppingCartDto actual = cartService.addToCart(requestDto, 1L);
```

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Nice! Some comments left

**Pull request merged**